### PR TITLE
[WIP] Allow to run SQL sandbox mode on multiple repos

### PR DESCRIPTION
**Pull request by AndrewDryga**
Sometimes you have more than one repo per application, in this cases SQL sandbox mode won't work since it expects only one encoded metadata object in `User-Agent` header.

I have extended it by allowing to pass multiple metadatas separated by comma:
```
config_meta = Phoenix.Ecto.SQL.Sandbox.metadata_for(Gateway.DB.Configs.Repo, self())
encoded_config = {:v1, config_meta}
|> :erlang.term_to_binary
|> Base.url_encode64

logger_meta = Phoenix.Ecto.SQL.Sandbox.metadata_for(Gateway.DB.Logger.Repo, self())
encoded_logger = {:v1, logger_meta}
|> :erlang.term_to_binary
|> Base.url_encode64

[{"user-agent", "BeamMetadata (#{encoded_config},#{encoded_logger})"}]
```

### CODE REVIEW

**Comment by AndrewDryga** [comment]
- [ ] Write tests
- [ ] Update docs

File: lib/phoenix_ecto/sql/sandbox.ex [hunk 1]
```diff
@@ -65,8 +65,12 @@ defmodule Phoenix.Ecto.SQL.Sandbox do
   defp extract_metadata(user_agent) when is_binary(user_agent) do
     ua_last_part = user_agent |> String.split("/") |> List.last
     case Regex.run(~r/BeamMetadata \((.*?)\)/, ua_last_part) do
-      [_, metadata] -> parse_metadata(metadata)
-      _             -> %{}
+      [_, metadata] ->
+        metadata
+        |> String.split(",")
+        |> Enum.reduce([], fn meta, acc -> [parse_metadata(meta)] ++ acc end)
```
Comment (josevalim) [inline lib/phoenix_ecto/sql/sandbox.ex]: Isn't this an Enum.map?

File: lib/phoenix_ecto/sql/sandbox.ex [hunk 2]
```diff
@@ -76,8 +80,10 @@ defmodule Phoenix.Ecto.SQL.Sandbox do
     |> Base.url_decode64!
     |> :erlang.binary_to_term
     |> case do
-         {:v1, metadata} -> metadata
-         _               -> %{}
+         {:v1, metadata} ->
+          metadata
```
Comment (josevalim) [inline lib/phoenix_ecto/sql/sandbox.ex]: Please avoid style changes on unrelated parts of the code during PRs.

**Comment by josevalim** [comment]
Sorry but I don't follow the need for this PR. `metadata_for` supports today the ability to pass multiple repositories:

    metadata_for [Gateway.DB.Configs.Repo, Gateway.DB.Logger.Repo], self())

Can't you use that instead?

**Comment by AndrewDryga** [comment]
Wow, did not notice it. Thanks a loot :)

**Pull request closed without merging**